x-pack/plugins/watcher/public/sections/watch_edit/components/monitoring_watch_edit/monitoring_watch_edit.tsx

```diff
@@ -16,13 +16,51 @@ import {
   EuiText,
   EuiLink,
 } from '@elastic/eui';
-import { i18n } from '@kbn/i18n';
 import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
 import { WatchContext } from '../../watch_context';
 
 const MonitoringWatchEditUi = ({ pageTitle }: { pageTitle: string }) => {
   const { watch } = useContext(WatchContext);
 
+  const editSystemWatchTitle = (
+    <FormattedMessage
+      id="xpack.watcher.sections.watchEdit.monitoring.edit.calloutTitleText"
+      defaultMessage="This watch cannot be edited."
+    />
+  );
+
+  const editSystemWatchMessage = (
+    <FormattedMessage
+      id="xpack.watcher.sections.watchEdit.monitoring.edit.calloutDescriptionText"
+      defaultMessage="Watch '{watchName}' is a system watch and cannot be edited. {watchStatusLink}"
+      values={{
+        watchName: watch.name,
+        watchStatusLink: (
+          <EuiLink href={`#/management/elasticsearch/watcher/watches/watch/${watch.id}/status`}>
+            <FormattedMessage
+              id="xpack.watcher.sections.watchEdit.monitoring.header.watchLinkTitle"
+              defaultMessage="View watch status."
+            />
+          </EuiLink>
+        ),
+      }}
+    />
+  );
+
+  const createSystemWatchTitle = (
+    <FormattedMessage
+      id="xpack.watcher.sections.watchEdit.monitoring.create.calloutTitleText"
+      defaultMessage="Cannot create watch"
+    />
+  );
+
+  const createSystemWatchMessage = (
+    <FormattedMessage
+      id="xpack.watcher.sections.watchEdit.monitoring.create.calloutDescriptionText"
+      defaultMessage="System watches are read-only and cannot be created."
```

Comment:
Out of curiosity why is `watcher/watches/new-watch/monitoring` a valid route? I couldn't find a way to navigate there via the UI, so would a better UX be one which handles this route the same way we handle other invalid routes?

![image](https://user-images.githubusercontent.com/1238659/59220947-00abad00-8b7b-11e9-8dc5-d0d17434b089.png)

Hmm well, maybe we want to handle invalid routes gracefully, e.g. by redirecting users back to the watch listing. :)

Reply:
good point :) 

it was valid bc it's a valid watch type in the code, but i adjusted the route definition to only allow the create route for threshold and json watch types. anything else will be redirected back to the watch list page.

```
<Route exact path={`${BASE_PATH}watches/new-watch/:type(json|threshold)`} component={WatchEdit} />
```